erdpy/docker.py

```diff
@@ -0,0 +1,57 @@
+import os
+import logging
+import subprocess
+from pathlib import Path
+from typing import List
+from erdpy.errors import DockerMissingError
+
+
+logger = logging.getLogger("build-with-docker")
+
+
+def is_docker_installed():
+    try:
+        result = subprocess.run(["docker", "--version"], capture_output=True, text=True)
+        output = result.stdout
+    
+        if "Docker version" in output:
+            return True
+        else:
+            raise DockerMissingError()
+    except:
+        logger.error("Something went wrong when checking if docker is installed!")
+
+
+def run_docker(image: str, project_path: Path, contract: str, output_path: Path, no_wasm_opt: bool):
+    docker_mount_args: List[str] = ["--volume", f"{output_path}:/output"]
+
+    if project_path:
+        docker_mount_args.extend(["--volume", f"{project_path}:/project"])
+
+
+    docker_args = ["docker", "run"]
+
+    docker_args += ["--interactive"]
+    docker_args += ["--tty"]
+    docker_args += docker_mount_args
+    docker_args += ["--user", f"{str(os.getuid())}:{str(os.getgid())}"]
```

Comment:
In the PR description, add a link to https://github.com/ElrondNetwork/elrond-sdk-images-build-contract-rust/blob/main/build_with_docker.py.